### PR TITLE
Feature 12285 environment jurisdiction

### LINKED ISSUE #12285: Do not allow the editing of environmental entities not owned outside the user's jurisdiction [1]
### Problem Description

Currently, for the environmental entities, as long as the users have the 'Edit existing environments' they can edit information on the entity page even when the entity is not owned by them and outside their jurisdiction.

### Proposed Change

No longer allow users to edit information on the environmental entity page if the entity is outside the user's jurisdiction and the entity is not owned by them.

### Added Value/Benefit

Consistency with the edit behavior on the other SORMAS entities.

### Acceptance Criteria

- [ ] make the environment entity page read only for users that do not own the entity, and the entity is outside the user's jurisdiction;
- [ ] when the user is the owner of the entity, and the entity is outside the user's jurisdiction keep allowing the editing;
- [ ] when the user is not the owner of the entity, and the entity is initially in their jurisdiction, but the user tries to change the jurisdiction of the entity show a warning pop-up upon saving (similar to the one we have for Events, for example); the pop-up should say: 
-The heading "Environment location update"; 
-Text of the pop-up "Changing the location of this environment could make you lose access to its details and/or disallow you to edit in the future. Are you sure you want to proceed?";
-Buttons: 'Yes' and 'No'.

### Implementation Details

_No response_

### Mockups

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sergiupacurariu**
Fixes #12285

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12420

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12420

File: sormas-app/app/src/main/java/de/symeda/sormas/app/util/JurisdictionHelper.java [hunk 1]
```diff
@@ -198,6 +200,41 @@ public static EventParticipantJurisdictionDto createEventParticipantJurisdiction
...
 	}
 
+	public static EnvironmentJurisdictionDto createEnvironmentJurisdictionDto(Environment environment) {
+		if (environment == null) {
+			return null;
+		}
+
+		Location environmentLocation = environment.getLocation();
+		if (environmentLocation == null) {
+			return null;
```
Comment (leventegal-she) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/util/JurisdictionHelper.java:210]: should not return null here

reporting and responsible users don't depend on the location

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-app/app/src/main/java/de/symeda/sormas/app/util/JurisdictionHelper.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/util/JurisdictionHelper.java]: updated the logic

**Review by leventegal-she — APPROVED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12420

**Pull request merged**